### ISSUE TITLE
Kill serde_json dependency

### ISSUE THREAD

**Issue by Manishearth** [servo/rust-bindgen#499]
bindgen currently pulls in serde_json (and with it a whole host of dependencies). The only reason we do this is because `target_build_utils` (a dependency of `libloading`) has a mode where you can give it a path to a json target spec.

It would be nice if this mode was made optional [using cargo features, specifically an on-by-default feature](http://doc.crates.io/manifest.html#the-features-section). It could be threaded through Cargo features and turned off when used in bindgen.

https://github.com/nagisa/target_build_utils.rs/blob/a05415e6dd366a869f030e4e2dd6a5b4fe8963df/src/lib.rs#L137 is where the json ge

cc @nagisa

(Filing the bug here instead of on target_build_utils so that someone might pick it up as an easy bug)

**Comment by mdevlamynck** [comment]
I can take this. I have wrote a patch for target_build_utils (I'll soon create the PR for it). But I'm not sure how to reflect this change in rust-bindgen though. We would need to also update libloading (to use the no-default-feature flag) right ?

**Comment by emilio** [comment]
Yes, or to adda feature in the same fashion, depending on whether
libloading uses it or not.

**Comment by mdevlamynck** [comment]
Ok, thx. I'll get working on the PRs for target_build_utils and libloading then.

**Comment by nagisa** [comment]
If there's any PRs I'll look at them on a weekend.

**Comment by nox** [comment]
Note that such a PR on target_build_utils would be a breaking change, and thus we will also need a correspond libloading bump.

**Comment by nox** [comment]
@Manishearth Even as a default feature, someone could be depending on it with `default-features = false`.

**Comment by nagisa** [comment]
I generally find it weird somebody would rather get rid of a dependency than just support people who use custom target specs.

That being said, there’s a nifty feature in nightly cargo which exposes the necessary data through environment variables, so eventually I’ll be able to both drop the dependency and keep the support for custom target specs. 

In the meantime, knock yourselves out with the 0.3.2 release (I hadn’t considered breaking change aspect on bumping the version number yesterday, so its a minor version bump)

**Comment by Manishearth** [comment]
Thanks!

**Comment by jdm** [comment]
What's the next step here since https://github.com/nagisa/target_build_utils.rs/issues/3 merged?

**Comment by jdm** [comment]
Hello? The issue is marked easy and assigned, and I can't figure out what its status actually is.

**Comment by Manishearth** [comment]
Unassigned since it's been a month. Feel free to ping and ask for assignment.

The next step is to add this default feature to libloading.

**Comment by nagisa** [comment]
Not sure libloading needs any changes.

**Comment by Manishearth** [comment]
Don't we need to be able to pass through the feature flag?

**Comment by TannerRogalsky** [comment]
@highfive assign me

**Comment by fitzgen** [comment]
@TannerRogalsky how are things going? Anything I can help with? Any questions I can answer for you?

**Comment by TannerRogalsky** [comment]
@fitzgen I got stuck trying to figure out how to disable default features on a dependency of a dependency. 

The relevant piece of documentation for reexporting features that I could find was [here](http://doc.crates.io/manifest.html#the-features-section).

```
# Features can be used to reexport features of other packages. The `session`
# feature of package `awesome` will ensure that the `session` feature of the
# package `cookie` is also enabled.
session = ["cookie/session"]
```

But I'm not sure how this idiom extends to `default-features = false`.

**Comment by nox** [comment]
> @fitzgen I got stuck trying to figure out how to disable default features on a dependency of a dependency.

If you want to disable the default features of A, which is depended upon by B in C, you need to disable them in B.

**Comment by nagisa** [comment]
libloading 0.4.0 does not depend on target_build_utils anymore.

**Comment by TannerRogalsky** [comment]
@nagisa Fantastic. That should simplify things, I think. I've updated clang-sys to point at your new version. After that is merged and bumped, I should be able to update rust-bindgen's dependency on clang-sys which should close this issue.

**Comment by TannerRogalsky** [comment]
This issue should be resolved with #685.

**Comment by fitzgen** [comment]
Thanks!
